server/src/test/java/org/opensearch/repositories/blobstore/BlobStoreRepositoryRemoteIndexTests.java

```diff
@@ -368,4 +378,239 @@ public void testRetrieveShallowCopySnapshotCase2() throws IOException {
         assertThat(snapshotIds, equalTo(originalSnapshots));
     }
 
+    public void testShallowAndFullCopyCreateAndDelete() throws IOException, InterruptedException {
+        FeatureFlagSetter.set(FeatureFlags.REMOTE_STORE);
+        final Client client = client();
+        final String snapshotRepositoryName = "test-repo";
+        final String remoteStoreRepositoryName = "test-rs-repo";
+
+        logger.info("-->  creating snapshot repository");
+
+        Settings snapshotRepoSettings = Settings.builder()
+            .put(node().settings())
+            .put("location", OpenSearchIntegTestCase.randomRepoPath(node().settings()))
+            .put(BlobStoreRepository.REMOTE_STORE_INDEX_SHALLOW_COPY.getKey(), Boolean.TRUE)
+            .build();
+        createRepository(client, snapshotRepositoryName, snapshotRepoSettings);
+
+        logger.info("-->  creating remote store repository");
+        Settings remoteStoreRepoSettings = Settings.builder()
+            .put(node().settings())
+            .put("location", OpenSearchIntegTestCase.randomRepoPath(node().settings()))
+            .build();
+        createRepository(client, remoteStoreRepositoryName, remoteStoreRepoSettings);
+
+        logger.info("--> creating an index and indexing documents");
+        final String indexName = "test-idx";
+        createIndex(indexName);
+        ensureGreen();
+        indexDocuments(client, indexName);
+
+        logger.info("--> creating a remote store enabled index and indexing documents");
+        final String remoteStoreIndexName = "test-rs-idx";
+        Settings indexSettings = getRemoteStoreBackedIndexSettings(remoteStoreRepositoryName);
```

Comment:
Given remote index is now enforced at cluster level we should cleanup this method. It is causing confusion that index created with these settings are remote indices vs with `getIndexSettings` are docRep based which is not the case here.

Reply:
will raise another refactoring PR for all the snapshot interop related tests. some other methods as well needs to be updated.

Reply:
If it is contained for the tests in this PR I would still encourage to make this change in this PR itself. Other tests can be updated separately.